can't eject: "Error running eject: undefined"

## Description

Brand new project, then run `npm run eject`

## Expected Behavior

I expected the to be able to eject (in order to add a native linked library) and keep the Expo libs included.

## Observed Behavior

What actually happened when you performed the above actions?

If there's an error message, please paste the *full terminal output and error message* in this code block:

```
Helios:k10 kimon$ npm run eject

> k10@0.1.0 eject /Users/kimon/Development/k10
> react-native-scripts eject


We didn't find any uses of the Expo SDK in your project, so you should be fine to eject to
"Plain" React Native. (This check isn't very sophisticated, though.)

We strongly recommend that you read this document before you proceed:
  https://github.com/react-community/create-react-native-app/blob/master/EJECTING.md

Ejecting is permanent! Please be careful with your selection.

? How would you like to eject from create-react-native-app? ExpoKit: I'll create or log in with an Expo account to use React Native and the Expo SDK.

An Expo account is required to proceed.

? It appears you're already logged in to Expo as kinergy, would you like to continue with this account? Yes, continue as kinergy.
Validating project manifest...
Downloading iOS code...
Moving iOS project files...
Error running eject: undefined
```

## Environment

Please run these commands in the project folder and fill in their results:

* `npm ls react-native-scripts`:
```
k10@0.1.0 /Users/kimon/Development/k10
└── react-native-scripts@0.0.29
```
* `npm ls react-native`:
```
k10@0.1.0 /Users/kimon/Development/k10
└── react-native@0.44.0
```
* `npm ls expo`:
```
k10@0.1.0 /Users/kimon/Development/k10
└─┬ expo@17.0.0
  └── UNMET PEER DEPENDENCY react@>=15.3.1

npm ERR! peer dep missing: react@>=15.3.1, required by lottie-react-native@1.1.1
npm ERR! peer dep missing: react@>=15.4.0, required by react-native-branch@2.0.0-beta.3
npm ERR! peer dep missing: react@>=15.4.0, required by react-native-maps@0.14.0
```
* `node -v`:
```
v7.10.0
```
* `npm -v`:
```
4.2.0
```
* `yarn --version`:
```
0.24.5
```
* `watchman version`:
```
{
    "version": "4.7.0"
}
```

Also specify:

1. Operating system: Mac OS X 10.12.5
2. Phone/emulator/simulator & version: N/A

## Reproducible Demo

Please provide a minimized reproducible demonstration of the problem you're reporting.

Issues that come with minimal repro's are resolved much more quickly than issues where a maintainer has to reproduce themselves.

--- a/VERSIONS.md
+++ b/VERSIONS.md
@@ -15,3 +15,4 @@ Each version of these dependencies is only compatible with a narrow version rang
 | 0.41.x         | 15.4.x         | 14.x.x | `"14.0.0"`               |
 | 0.42.x         | 15.4.x         | 15.x.x | `"15.0.0"`               |
 | 0.43.x         | 16.0.0-alpha.6 | 16.x.x | `"16.0.0"`               |
+| 0.44.x         | 16.0.0-alpha.6 | 17.x.x | `"17.0.0"`               |

--- a/react-native-scripts/package.json
+++ b/react-native-scripts/package.json
@@ -1,6 +1,6 @@
 {
   "name": "react-native-scripts",
-  "version": "0.0.29",
+  "version": "0.0.30",
   "description": "Configuration and scripts for Create React Native App.",
   "license": "BSD-3-Clause",
   "keywords": [
@@ -35,7 +35,7 @@
     "path-exists": "^3.0.0",
     "progress": "^2.0.0",
     "qrcode-terminal": "^0.11.0",
-    "xdl": "^40.0.0"
+    "xdl": "^41.0.0"
   },
   "devDependencies": {
     "babel-plugin-add-module-exports": "^0.2.1",

--- a/react-native-scripts/src/scripts/init.js
+++ b/react-native-scripts/src/scripts/init.js
@@ -9,14 +9,14 @@ import log from '../util/log';
 
 // UPDATE DEPENDENCY VERSIONS HERE
 const DEFAULT_DEPENDENCIES = {
-  expo: '^16.0.0',
+  expo: '^17.0.0',
   react: '16.0.0-alpha.6',
-  'react-native': '^0.43.4',
+  'react-native': '^0.44.0',
 };
 
 // TODO figure out how this interacts with ejection
 const DEFAULT_DEV_DEPENDENCIES = {
-  'jest-expo': '^0.4.0',
+  'jest-expo': '~1.0.1',
   'react-test-renderer': '16.0.0-alpha.6',
 };
 

--- a/react-native-scripts/template/app.json
+++ b/react-native-scripts/template/app.json
@@ -1,5 +1,5 @@
 {
   "expo": {
-    "sdkVersion": "16.0.0"
+    "sdkVersion": "17.0.0"
   }
 }

